controller/src/main/java/io/pravega/controller/store/PravegaTablesStoreHelper.java

```diff
@@ -244,6 +248,101 @@ public void invalidateCache(String table, String key) {
                 .whenComplete((r, ex) -> releaseEntries(entries));
     }
 
+    /**
+     * Remove value of every entry for the provided keys.
```

Comment:
this logic should be moved out.
the caller should use an updateValues method which you can add here. 
where they get and update all the lists and issue a conditional update on the helper. 

alternatively, you can pass an update lambda.

getAndUpdateValues(final String tableName, Set<String> tableKeys, 
                                                BiFunction<String, V, V> updateFunction,
                                                Function<byte[], V> fromBytes,
                                                long requestId)
logically following can be done:
1. Map<K+keyVersion, V> vbytes = getValues(tableKeys);
2. updates = foreach vbyte: vBytes -> update = updateFunction(fromBytes(vByte))
3. updateValues(k+keyVersion, updates);

Reply:
presently here the helper is assuming that the values are a list of string and one entry from the list has to be removed.

Reply:
Thanks, I have improved the abstraction now. Let me know.